### DISCUSSION TITLE
Twill and SSO

### DISCUSSION THREAD

**Discussion by ghost** [area17/twill#1286] [Q&A] [answered]
Hi again,

Is there a way to integrate SSO with Twill login/authentication? Ive got some simplesamlephp stuff set up for my project and was just wondering if there was an easy way to use SSO to log in to the Twill CMS

Cheers,
Aaron

**Comment by ifox** [comment, marked as answer]
Hi @aarontilt,

Yes it is definitely possible. The process would currently involve adding a link to your SSO login from Twill's login page, by creating ` resources/views/vendor/twill/auth/login.blade.php`. Then, you can listen to the SAML login/logout events to find or create Twill users and authenticate them programmatically.
